packages/core/micro/test/matrix.spec.ts

```diff
@@ -95,8 +102,8 @@ export class TestMatrix<T = any, TRow = never, TCol = never> implements IMatrixC
 
     public constructor (
         producer: IMatrixProducer<T>,
-        private readonly rowWriter: IVectorWriter<TRow>,
-        private readonly colWriter: IVectorWriter<TCol>,
+        private readonly rowWriter: IVectorWriter<TRow> & IVectorShapeWriter,
+        private readonly colWriter: IVectorWriter<TCol> & IVectorShapeWriter,
```

Comment:
Writers can overwrite existing values.
ShapeWriters can move, remove, and insert empty items.

In general, if you need to know <T> for the operation, it's a write.
If you can do the operation w/o knowing T, it's a shape change.